spec/save_inventory/strategies_and_references_spec.rb

```diff
@@ -139,39 +141,34 @@
           end
 
           it "tests that a key pointing to a relation is filled correctly when coming from db" do
-            vm_refs               = ["vm_ems_ref_3", "vm_ems_ref_4"]
-            network_port_refs     = ["network_port_ems_ref_1"]
+            vm_refs               = %w(vm_ems_ref_3 vm_ems_ref_4)
+            network_port_refs     = %w(network_port_ems_ref_1)
 
             # Setup InventoryCollections
-            @data                 = {}
-            @data[:network_ports] = ::InventoryRefresh::InventoryCollection.new(
-              network_ports_init_data(
-                :parent   => @ems.network_manager,
-                :arel     => @ems.network_manager.network_ports.where(:ems_ref => network_port_refs),
-                :strategy => :local_db_find_missing_references
-              )
+
+            network_ports_init_data(
+              :parent   => @ems.network_manager,
+              :arel     => @ems.network_manager.network_ports.where(:ems_ref => network_port_refs),
+              :strategy => :local_db_find_missing_references
             )
-            @data[:vms] = ::InventoryRefresh::InventoryCollection.new(
-              vms_init_data(
-                :arel     => @ems.vms.where(:ems_ref => vm_refs),
-                :strategy => :local_db_find_missing_references
-              )
+
+            vms_init_data(
+              :arel     => @ems.vms.where(:ems_ref => vm_refs),
+              :strategy => :local_db_find_missing_references
             )
-            @data[:hardwares] = ::InventoryRefresh::InventoryCollection.new(
-              hardwares_init_data(
-                :arel     => @ems.hardwares.joins(:vm_or_template).where(:vms => {:ems_ref => vm_refs}),
-                :strategy => db_strategy
-              )
+            hardwares_init_data(
+              :arel     => @ems.hardwares.joins(:vm_or_template).where(:vms => {:ems_ref => vm_refs}),
+              :strategy => db_strategy
             )
 
             # Parse data for InventoryCollections
             @network_port_data_1 = network_port_data(1).merge(
-              :device => @data[:hardwares].lazy_find(@data[:vms].lazy_find(vm_data(1)[:ems_ref]), :key => :vm_or_template)
+              :device => @persister.hardwares.lazy_find(@persister.vms.lazy_find(vm_data(1)[:ems_ref]), :key => :vm_or_template)
             )
 
             # Fill InventoryCollections with data
-            add_data_to_inventory_collection(@data[:network_ports],
-                                             @network_port_data_1)
+            add_data_to_persisters_collection(@persister, :network_ports,
```

Comment:
we can probably do just the persister.collection.build(data), like we do in persister specs?

Reply:
this method does exactly the same, but it's much easier to rewrite almost only method name

Reply:
..and much more transparent to review, I think

Reply:
right, we should rewrite it to be consistent with other as follow up PR then